test/path_matcher_test.cc

```diff
@@ -819,6 +823,37 @@ TEST_F(PathMatcherTest, VariableBindingsWithQueryParamsEncoding) {
             bindings);
 }
 
+TEST_F(PathMatcherTest, QueryParameterNotUnescapePlus) {
+  MethodInfo* a = AddGetPath("/a");
+  Build();
+
+  EXPECT_NE(nullptr, a);
+
+  Bindings bindings;
+  EXPECT_EQ(LookupWithParams("GET", "/a", "x=Hello+world&y=%2B+%20", &bindings), a);
+  EXPECT_EQ(Bindings({
+                Binding{FieldPath{"x"}, "Hello+world"},
+                Binding{FieldPath{"y"}, "++ "},
+            }),
+            bindings);
+}
+
+TEST_F(PathMatcherTest, QueryParameterUnescapePlus) {
+  MethodInfo* a = AddGetPath("/a");
+  SetQueryParamUnescapePlus(true);
+  Build();
+
+  EXPECT_NE(nullptr, a);
+
+  Bindings bindings;
+  EXPECT_EQ(LookupWithParams("GET", "/a", "x=Hello+world&y=%2B+%20", &bindings), a);
+  EXPECT_EQ(Bindings({
+                Binding{FieldPath{"x"}, "Hello world"},
+                Binding{FieldPath{"y"}, "+  "},
+            }),
+            bindings);
+}
+
```

Comment:
Please add a test to ensure `+` is not unescaped when in the path portion of the URL.

Reply:
Done